airbyte-ci/connectors/pipelines/pipelines/actions/environments.py

```diff
@@ -470,13 +471,18 @@ def with_global_dockerd_service(dagger_client: Client) -> Container:
         Container: The container running dockerd as a service
     """
     return (
-        dagger_client.container()
-        .from_(consts.DOCKER_DIND_IMAGE)
-        .with_mounted_cache(
-            "/tmp",
-            dagger_client.cache_volume("shared-tmp"),
-        )
+        dagger_client.container().from_(consts.DOCKER_DIND_IMAGE)
+        # We set this env var because we need to use a non-default zombie reaper setting.
+        # The reason for this is that by default it will want to set its parent process to 1 when reaping.
+        # This won't be possible because of container-ception: dind is running inside the dagger engine.
+        .with_env_variable("TINI_SUBREAPER", "")
+        # Expose the docker host port.
         .with_exposed_port(2375)
+        # Mount the docker cache volumes.
+        .with_mounted_cache("/tmp", dagger_client.cache_volume("shared-tmp"))
+        # Mount /var/lib/docker so docker writes to the host file system, i.e. the dagger engine.
+        .with_mounted_cache("/var/lib/docker", dagger_client.cache_volume("shared-var-lib-docker"))
```

Comment:
In this case all we want is this to be a volume, to avoid storage engine issues in dind. We don't actually care about the caching functionality. Perhaps we can add an `rm -rf /var/lib/docker` in the with_exec below to nuke the cache?

Reply:
> avoid storage engine issues in dind

These were the reason of other errors/warning logs from dockerd right?

> Perhaps we can add an rm -rf /var/lib/docker in the with_exec below to nuke the cache?

As it's run as a service container the last with_exec should be a never ending process (the `dockerd` command).  Moreover if the dockerd command crashes with a `1` status code the `rm` execution will never happen.  
We could add the `rm -rf /var/lib/docker` as a first `with_exec`,  before the `dockerd` with exec to start `dockerd` on a clean env.

Reply:
Yes. I think I witnessed instances of this corruption. I'm going to try using a different storage engine, fuse-overlayfs. I have no idea what I'm doing.